src/libraries/Common/src/System/Net/Http/X509ResourceClient.cs

```diff
@@ -97,7 +98,11 @@ internal static partial class X509ResourceClient
                     return null;
                 }
 
-                Type taskOfHttpResponseMessageType = typeof(Task<>).MakeGenericType(httpResponseMessageType);
+                [UnconditionalSuppressMessage("AotAnalysis", "IL3050:RequiresDynamicCode",
+                   Justification = "The type, httpResponseMessageType, will be available during runtime")]
```

Comment:
This is not the right justification. The right justification is that that this `MakeGenericType` call is creating an instantiation over reference type.

Reply:
Ah, I think the idea is that this is return type of existing method is preserved and thus it has to be available at runtime. Is that right?

Reply:
It should be possible to write this without any AOT suppressions: You can call `Type.GetType` on the generic type that this is trying to instantiate. It should not need any warning suppressions and it should also fix the build errors that the PR is failing with.

Reply:
Is this analyzer-only warning?
I would expect the data flow in ilc to figure this out without any warning/suppression, but the analyzer probably won't.

Reply:
Unfortunately, the `Type.GetType` on the generic type doesn't work likely due `ILCompiler` not yet having done this work. Will change the suppression message with the right justification and push. 

I haven't been able to repro the build failures below locally :-(

Reply:
We don't analyze arguments to MakeGenericType/MakeGenericMethod because they're rarely statically analyzable. We could, but I don't expect it to kick in often. Normally people would just do `typeof(Foo<Bar>)` instead of `typeof(Foo<>).MakeGenericType(typeof(Bar))`, unless they have weird layering issues like the code here.

Reply:
> Unfortunately, the Type.GetType on the generic type doesn't work likely due ILCompiler not yet having done this work

Do we have a bug on this? If not, we should get one.

I have done a quick test compiling the following with `PublishAot=true. It does not produce any warnings, but it does not work as expected at runtime.

```
Type? t = Type.GetType("System.Collections.Generic.List`1[[System.DayOfWeek, System.Runtime]], System.Collections");
Console.WriteLine(t != null);
```

Reply:
Ah, I see what you mean - MakeGenericType is not needed for that one.

This was tracked in the abandoned https://github.com/dotnet/runtimelab/issues/339 issue.

I filed a new one as https://github.com/dotnet/runtime/issues/72833 for this specific thing.